src/bowling.js

```diff
@@ -0,0 +1,90 @@
+function strict(){}
+
+function Bowling() {
+  this._frames = [];
+  this.frameScore = [];
+  this._score = 0;
+}
+
+Bowling.prototype.throwBall = function(pins) {
+  this.frameScore.push(pins);
+  if (this.frameScore.length == 2 || this.frameScore[0] == 10){
+    this._frames.push(this.frameScore);
+    this.frameScore = [];
+  }
+};
+
+Bowling.prototype.currentFrame = function() {
+  return this._frames.length + 1;
+}
+
+Bowling.prototype.calculateScore = function() {
+  var self = this;
+  self._score = 0;
+  if (self._frames.length > 0){
+    self._frames.forEach(function(frames,i){
+      if(self.isStrike(frames)){
+        self.strikeScore(i)
+      }else if(self.isSpare(frames)){
+          self.spareScore(i)
+        }else{
```

Comment:
pyramid of doom! nested if/else conditional will cause headaches when you need to make any changes to this class.